salt/modules/dockermod.py

```diff
@@ -3807,9 +3808,12 @@ def rm_(name, force=False, volumes=False, **kwargs):
             'remove this container'.format(name)
         )
     if stop_ and not force:
+        auto_remove = inspect_container(name)['HostConfig']['AutoRemove']
```

Comment:
To future-proof us against Docker changing inspect return data structure, we should catch on `KeyError` here:

```python
        inspect_results = inspect_container(name)
        try:
            auto_remove = inspect_results['HostConfig']['AutoRemove']
        except KeyError:
            log.error(
                'Failed to find AutoRemove in inspect results, Docker API may '
                'have changed. Full results: %s', inspect_results
            )
            # Assume False and attempt to remove the container below
            auto_remove = False
```

Reply:
agreed with eafp

Reply:
Good point, I forgot you had set it above. :+1: